drivers/ethernet/Kconfig.nxp_enet

```
@@ -0,0 +1,52 @@
+# NXP ENET Ethernet driver configuration options
+
+# Copyright 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig ETH_NXP_ENET
+	bool "NXP ENET Ethernet driver"
+	default y
+	depends on DT_HAS_NXP_ENET_MAC_ENABLED
+	select NOCACHE_MEMORY if HAS_MCUX_CACHE
+	select ARM_MPU if CPU_CORTEX_M7
+	select MDIO if DT_HAS_NXP_ENET_MDIO_ENABLED
+	select EXPERIMENTAL
+	help
+	  Enable NXP ENET Ethernet driver.
+
+if ETH_NXP_ENET
+
+config ETH_NXP_ENET_HW_ACCELERATION
+	bool "Hardware acceleration"
+	help
+	  Enable hardware acceleration for the following:
+	  - IPv4, UDP and TCP checksum (both Rx and Tx)
```

Add a dependency if it only works for IPv4.

```suggestion
config ETH_NXP_ENET_HW_ACCELERATION
	bool "Hardware acceleration"
	default y
	depends on !NET_IPV6
	help
	  Enable hardware acceleration for the following:
	  - IPv4, UDP and TCP checksum (both Rx and Tx)
```

---

But if you have both IPv4 and IPv6 enabled, then the HW acceleration would be disable for IPv4 too. Whatever check there is, it should be done at runtime.